client/state/sites/selectors.js

```diff
@@ -86,14 +85,12 @@ export const getSite = createSelector(
 		// To avoid mutating the original site object, create a shallow clone
 		// before assigning computed properties
 		site = { ...site };
-		site.hasConflict = isSiteConflicting( state, siteId );
-		assign( site, getComputedAttributes( site ) );
+		assign( site, getSiteComputedAttributes( state, siteId ) );
 		assign( site, getJetpackComputedAttributes( state, siteId ) );
-		site.is_previewable = isSitePreviewable( state, siteId );
 
 		return site;
 	},
-	( state ) => state.sites.items
+	( state, siteId ) => [ get( state, [ 'sites', 'items', siteId ] ), get( state, [ 'currentUser', 'capabilities', siteId ] ) ]
```

Comment:
Our current test cases do not pass state information for functions that now require it. If get was not used, 44 test cases failed. To make them pass is easy we just need to pass the userState but that would make this PR very very huge. If we decide to update the test cases in order to not use this get. I can do some pull requests that update them. And than change this one.